### PR TITLE
List assets stories

### PR DESCRIPTION
**Pull request by abdullahtariqbutt**
Task 5 and 6

- List Stories and Assets
- Only allow owners to delete or edit there assets , stories and images

### CODE REVIEW

File: app/controllers/map_assets_controller.rb [hunk 1]
```diff
@@ -19,7 +21,11 @@ def create
 
     respond_to do |format|
       format.html do 
-        if success; redirect_to @map_asset, notice: "Asset was successfully created." else render :new end
+        if success
```
Comment (asad-imtiaz) [inline app/controllers/map_assets_controller.rb:24]: remove trailing whitespaces

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -10,4 +10,15 @@ def set_random_user_name()
     c = subsets.sample(length).join
     "username" + c
   end
+
+  STUFFTYPE = ["Good", "Bad", "Fix"]
```
Comment (asad-imtiaz) [inline app/helpers/application_helper.rb]: move these variables in the model and `.freeze`

File: app/helpers/stories_helper.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+module StoriesHelper
```
Comment (asad-imtiaz) [inline app/helpers/stories_helper.rb]: remove this file in not used

File: app/javascript/packs/rating.js [hunk 4]
```diff
@@ -16,25 +42,20 @@ window.rating_binder = function() {
     $(this).text('I don\'t want to leave a rating');
     $("#rate").hide("slow");
 
-    $('#rating').raty({
+    $('#rate #rating').raty({
```
Comment (asad-imtiaz) [inline app/javascript/packs/rating.js:45]: make it a class if it is being reused on different elements

File: app/services/average.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
+class Average
```
Comment (asad-imtiaz) [inline app/services/average.rb]: refactor as discussed

File: app/views/map_assets/_form.html.erb [hunk 6]
```diff
@@ -25,18 +20,12 @@
 
     <div class="field form-group">
       <%= form.label :category %>
-      <%= form.select :category, ["Art", "Education", "Health", "Business", "Enviroment", "Housing", "Child Care", "Food", "Industry", "Safety", "Transportation"] %>  
+      <%= form.select :category, get_categories() %>
```
Comment (asad-imtiaz) [inline app/views/map_assets/_form.html.erb]: don't need `()`

File: app/views/map_assets/_form.html.erb [hunk 7]
```diff
@@ -25,18 +20,12 @@
...
     <div class="field form-group">
       <%= form.label :stuff_type %>
-      <br>
-      <%= form.radio_button :stuff_type, "Good" %>
-      <%= form.label :Good %>
-      <%= form.radio_button :stuff_type, "Bad" %>
-      <%= form.label :Bad %>
-      <%= form.radio_button :stuff_type, "Fix" %>
-      <%= form.label :Fix %>
+      <%= form.select :stuff_type, get_stuff_type() %>
```
Comment (asad-imtiaz) [inline app/views/map_assets/_form.html.erb]: dont need `()`

File: app/views/map_assets/_map_asset_row.html.erb [hunk 8]
```diff
@@ -1,11 +1,20 @@
...
   <td><%= asset.rating %></td>
   <td><%= asset.stuff_type %></td>
   <td><%= asset.description %></td>
   <td><%= link_to 'Show', map_asset_path(asset) %></td>
-  <td><%= link_to 'Edit', edit_map_asset_path(asset) %></td>
-  <td><%= link_to 'Destroy', map_asset_path(asset), method: :delete, data: { confirm: 'Are you sure?' } %></td>
+
+  <!-- User Sign In Check to Allow Edit and Delete -->
+  <% if user_signed_in? %>
+    <% if asset.user == current_user %>
```
Comment (asad-imtiaz) [inline app/views/map_assets/_map_asset_row.html.erb]: refactor this keeping in mind that if a user is not signed in, `current_user` returns nil

File: app/views/map_assets/show.html.erb [hunk 9]
```diff
@@ -11,45 +10,84 @@
...
-    <% end %>
-  <% end %>
-</div>
+<%= render "shared/display_uploads", target: @map_asset %>
 
 <br>
 
-<%= link_to 'Edit', edit_map_asset_path(@map_asset), class:"btn btn-primary btn-lg" %> |
+<!-- User Sign In Check to Allow Edit -->
+<% if user_signed_in? %>
```
Comment (asad-imtiaz) [inline app/views/map_assets/show.html.erb]: refactor

File: app/views/shared/_story_row.html.erb [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+  <td><%= story.name %></td>
+  <td><%= story.address %></td>
+  <td><%= story.category %></td>
+  <td><%= story.rating %></td>
+  <td><%= story.stuff_type %></td>
+  <td><%= story.description %></td>
+  <td><%= link_to 'Show', story %></td>
+
+  <!-- User Sign In Check to Allow Edit and Delete -->
+  <% if user_signed_in? %>
```
Comment (asad-imtiaz) [inline app/views/shared/_story_row.html.erb]: refactor

File: app/views/stories/create.js.erb [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+  rating_binder();
+
+<% else %>
+
+  $('#map_asset #rating_display').empty();
+  $('#map_asset #rating_display').data("score", <%= @story.map_asset.rating %>);
+   asset_rating();
+
+  $("#stories_table").prepend("<%= j render('shared/story_row', story: @story) %>");
+  $(".close-modal").click();
```
Comment (asad-imtiaz) [inline app/views/stories/create.js.erb]: find a way to close a modal through bootstrap Js functions

File: app/controllers/map_assets_controller.rb (hunk 1 above)
Comment (abdullahtariqbutt) [inline app/controllers/map_assets_controller.rb:24]: Done

File: app/helpers/application_helper.rb (hunk 2 above)
Comment (abdullahtariqbutt) [reply app/helpers/application_helper.rb]: Two models use this so, I have placed it in helper method and used .freeze with them,
Dynamic constant assignment error was showing , so it put it outside the functions

File: app/helpers/stories_helper.rb (hunk 3 above)
Comment (abdullahtariqbutt) [reply app/helpers/stories_helper.rb]: Done

File: app/javascript/packs/rating.js (hunk 4 above)
Comment (abdullahtariqbutt) [reply app/javascript/packs/rating.js:45]: Done

File: app/services/average.rb (hunk 5 above)
Comment (abdullahtariqbutt) [reply app/services/average.rb]: Done

File: app/views/map_assets/_form.html.erb (hunk 7 above)
Comment (abdullahtariqbutt) [reply app/views/map_assets/_form.html.erb]: Done

File: app/views/map_assets/_form.html.erb (hunk 6 above)
Comment (abdullahtariqbutt) [reply app/views/map_assets/_form.html.erb]: Done

File: app/views/map_assets/_map_asset_row.html.erb (hunk 8 above)
Comment (abdullahtariqbutt) [reply app/views/map_assets/_map_asset_row.html.erb]: Done

File: app/views/map_assets/show.html.erb (hunk 9 above)
Comment (abdullahtariqbutt) [reply app/views/map_assets/show.html.erb]: Done

File: app/views/shared/_story_row.html.erb (hunk 10 above)
Comment (abdullahtariqbutt) [reply app/views/shared/_story_row.html.erb]: Done

File: app/views/stories/create.js.erb (hunk 11 above)
Comment (abdullahtariqbutt) [reply app/views/stories/create.js.erb]: Done

**Pull request merged**